types/amazon-cognito-auth-js/index.d.ts

```diff
@@ -573,5 +590,83 @@ export class StorageHelper {
      * This is used to return the storage
      * @returns the storage
      */
-    getStorage(): Storage;
+    getStorage(): CognitoAuthStorage;
+}
+
+export interface CognitoAuthStorage {
+    /**
+     * Sets the value of the pair identified by key to value, creating a new
+     * key/value pair if none existed for key previously.
+     */
+    setItem(key: string, value: string): void;
+
+    /**
+     * Returns the current value associated with the given key, or null or
+     * undefined if the given key does not exist in the list associated with
+     * the object.
+     */
+    getItem(key: string): string | null | undefined;
```

Comment:
Stil not sure what case this can be `null`, but this will at least enforce type safety I think

Reply:
The intent in the package was for this interface to be compatible with WebStorage's `Storage`. The only snag is that `CookieStorage` returns `null` instead of `undefined`. To allow WebStorage implementations (e.g. `localStorage`) as well as `CookieStorage` to satisfy this interface the signature must allow both `null` and `undefined`.